time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateTimeArbitrary.java

```diff
@@ -0,0 +1,151 @@
+package net.jqwik.time.internal.properties.arbitraries;
+
+import java.time.*;
+import java.time.temporal.*;
+
+import org.apiguardian.api.*;
+
+import net.jqwik.api.*;
+import net.jqwik.api.arbitraries.*;
+import net.jqwik.time.api.*;
+import net.jqwik.time.api.arbitraries.*;
+
+import static java.time.temporal.ChronoUnit.*;
+import static org.apiguardian.api.API.Status.*;
+
+@API(status = INTERNAL)
+public class DefaultLocalDateTimeArbitrary extends ArbitraryDecorator<LocalDateTime> implements LocalDateTimeArbitrary {
+
+	private static final LocalDateTime DEFAULT_MIN = LocalDateTime.of(DefaultLocalDateArbitrary.DEFAULT_MIN_DATE, LocalTime.MIN);
+	private static final LocalDateTime DEFAULT_MAX = LocalDateTime.of(DefaultLocalDateArbitrary.DEFAULT_MAX_DATE, LocalTime.MAX);
+
+	private LocalDateTime min = null;
+	private LocalDateTime max = null;
+
+	private ChronoUnit ofPrecision = DefaultLocalTimeArbitrary.DEFAULT_PRECISION;
+
+	@Override
+	protected Arbitrary<LocalDateTime> arbitrary() {
+
+		LocalDateTime effectiveMin = calculateEffectiveMin();
+		LocalDateTime effectiveMax = calculateEffectiveMax();
+		if (effectiveMax.isBefore(effectiveMin)) {
+			throw new IllegalArgumentException("The maximum date time is too soon after the minimum date time.");
+		}
+
+		LocalDateArbitrary dates = Dates.dates();
+		LocalTimeArbitrary[] times = generateTimeArbitrary(effectiveMin, effectiveMax, ofPrecision);
```

Comment:
Don't use arrays unless forced by libraries to do so. Use list, tuples or maybe specialized container that shows what the individual elements mean.